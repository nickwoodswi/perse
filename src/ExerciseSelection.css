.exercise-selector{
    height: 70px;
    width: 290px;
    margin-right: 5px;
    border: none;
    font-size: 16px;
    padding: 5px;
    border-radius: 5px;
    box-shadow: 2px 2px 2px darkgrey;
}
