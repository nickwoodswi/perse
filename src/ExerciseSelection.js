import React, { Component } from 'react'
import './ExerciseSelection.css'

class ExerciseSelection extends Component {
    render() {
        if (this.props.type === 'select') {
            return(
                <select className="exercise-selector" onChange={this.props.setSelectionAsExercise}>
                    {this.props.selectorOptions.map((option, idx) => {
                        return(
                            <option key={idx} value={option.exercise_types_id}>{option.exercise_types_name}</option>
                        )
                    })}
                </select>
            )
        }
        if (this.props.type === 'create') {
            return(
                <input 
                    className='exercise-selector'
                    onChange={this.props.define}
                    id="ex-name-input" 
                    type="text" 
                    value={this.props.name} 
                    placeholder="New Exercise Name" />
            )
        }
    }
}

export default ExerciseSelection