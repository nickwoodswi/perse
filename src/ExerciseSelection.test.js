import React from 'react';
import { render } from '@testing-library/react';
import ExerciseSelection from './ExerciseSelection';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders ExerciseSelection component', () => {
    render(<MemoryRouter><ExerciseSelection /></MemoryRouter>);
});

it('ExerciseSelection renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><ExerciseSelection /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});