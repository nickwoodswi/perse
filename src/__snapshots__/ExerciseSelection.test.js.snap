// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseSelection renders the UI as expected 1`] = `
<select
  className="exercise-selector"
>
  <option>
    SELECT EXERCISE
  </option>
</select>
`;
